learn/sherif/main_test/src/main/java/frc/robot/MockableSparkMax.java

```diff
@@ -0,0 +1,24 @@
+package frc.robot;
+
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANSparkMax;
+
+public class MockableSparkMax extends CANSparkMax implements CANSparkFunctions {
+    /**
+     * Create a new SPARK MAX Controller
+     *
+     * @param deviceID The device ID.
+     * @param type     The motor type connected to the controller. Brushless motors
+     *                 must be connected to their matching color and the hall sensor
+     *                 plugged in. Brushed motors must be connected to the Red and
+     */
+    public MockableSparkMax(int deviceID, MotorType type) {
+        super(deviceID, type);
+
+    }
+
+    @Override
+    public CANEncoder GetEncoder() {
```

Comment:
Instead of using a slightly different name, does it work to use the same name?  So `CANSparkFunctions` has a function `getEncoder`?  That way we won't have to keep track of any differences in between the documentation on the spark max vs the our custom wrapper.  

Take a look at:
https://stackoverflow.com/questions/23946536/extended-class-and-implemented-interface-having-same-method-name

That suggests if the method as the same name we won't even have to implement it in the implementation class.